tests/profiler/bob_user_workflow_verbose_profiler_config.yml

```diff
@@ -0,0 +1,216 @@
+      class_name: SimpleColumnSuffixDomainBuilder
+      module_name: great_expectations.profiler.domain_builder.simple_column_suffix_domain_builder
+      column_name_suffixes:
+        - _id
+        - _type
+    parameter_builders:
+      - name: my_value_set_for_ids
+        # This parameter is the union of all observed batch value_sets
+        # MultiBatchParameterBuilder does not aggregate Parameters, it returns a list of value_set for each batch
+        class_name: MultiBatchParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.multi_batch_parameter_builder
+        metric_domain_kwargs: $domain.domain_kwargs
+        # Specify all batches prior to the 6 most recent to be used to create this set of Parameters using the below batch_request
+        batch_request:
+          partition_request:
+            partition_index: ":-6"
+        # TODO: Should we also support "offset" from the active_batch (active_batch can be set via passing a validator to the Profiler.profile() method and previously setting the active batch validator.active_batch_id(batch_id="mybatchid"))?
+        # Note: To specify all batches explicitly use the below batch_request
+        # batch_request:
+        #  partition_request:
+        #    partition_index: "0:"
+        # TODO: Maybe this can be omitted and inferred as the default (all batches)? James: maybe default isn't necessarily what's in the validator.
+        metric_name: column.distinct_values
+    expectation_configuration_builders:
+      # Create union of value sets here in the expectation_configuration_builder
+      - expectation_type: expect_column_values_to_be_in_set
+        column: $domain.domain_kwargs.column
+        # Parameter builder returns a list of value_sets so take the union() here
+        # TODO: Build `union(sets: List[Set])` method in GE parsing language
+        # TODO: How to reference list of parameters from ParameterBuilder (proposal below)?
+        value_set: union($parameter.my_value_set_for_ids.parameter_list)
+
+  my_rule_for_amounts:
+    # This rule creates expect_column_values_to_be_between & expect_column_mean_to_be_between for each of the _amount column domains
+    # Rationale: I want alerts when my current data varies from prior data (within some bounds)
+    class_name: Rule
+    module_name: great_expectations.profiler.rule.rule
+    domain_builder:
+      # Build list of domains from "amount" column domains: fare_amount,tip_amount,tolls_amount,total_amount
+      class_name: SimpleColumnSuffixDomainBuilder
+      module_name: great_expectations.profiler.domain_builder.simple_column_suffix_domain_builder
+      column_name_suffixes:
+        - _amount
+    parameter_builders:
+      - name: my_mean_parameter_for_batches_in_previous_full_year
+        # Parameter computes the mean of all batches in the previous full year
```

Comment:
I'm trying to place this in relation to the my_value_set_for_ids parameter for my_rule_for_ids. In my_value_set_for_ids, the parameter is a list of value_sets. We then use a union in the expect_configuration_builder to get a new metric/parameter/thing (the set containing all of these value_sets). In this case, the aggregation is done in the expectation_configuration_builder.

In this case, the output of the parameter is the mean of means. We do the aggregation in the parameter_builder. 

Is that right? Is there a reason why we would want to do the aggregation in the expectation_configuration_builder in one case, and in the parameter_builder in the other case? Do we specifically want to allow that flexibility?

Reply:
Ah - I was purposely trying to show both examples in this fixture (along with the third example of multiple parameters). Since both ParameterBuilders and ExpectationConfigurationBuilders are extensible I think both are valid configurations. If we want to steer toward mostly using ExpectationConfigurationBuilders for handling complex logic we can do that in the docs / examples. But I think the final call on whether this is a valid config is up to the product team.

Reply:
@anthonyburdi @talagluck I believe that `parameter_builders` now require `parameter_name` (as opposed to `name`).

Also, I believe that it is correct to say that in a lot of cases, aggregation will be done in an `ExpectationConfigurationBuilder` subclass.  This is because the `build_expectation_configuration` loop is run after all the parameters have been built and are thus available.  All the batches can be processed in the parameter builders, parameters set, and then parameters queried and their values summarized/aggregated in `ExpectationConfigurationBuilder`.  It would be really interesting to try to implement a few use cases and gauge the viability and degree of convenience of this "dev loop".

Reply:
@alexsherstinsky 1. yes - thanks for the catch! 2. I think this is a really valid point that given the current design of calculating metrics as parameters are built a user would have to make sure to order their `ParameterBuilders` carefully if they reference other parameters. In the case where a `ParameterBuilder` does it's own aggregation, the ordering is not so important.
My unsubstantiated feeling (could be wrong) is that in general we want to steer toward any aggregation in `ExpectationConfigurationBuilder`s but leave an escape hatch with custom `ParameterBuilder`s.

Reply:
@anthonyburdi FWIW, I feel the same (re aggregations should be in ECB).